core/Cmd.ml

```diff
@@ -139,6 +141,35 @@ let verbose_run_term : bool Term.t =
   in
   Arg.value (Arg.flag info)
 
+(* Converter for arguments of the form KEY=VALUE *)
+let env_conv =
+  let key_re = Re.Pcre.regexp {|\A[A-Za-z_][A-Za-z_0-9]*\z|} in
+  let error str = Error (sprintf "Malformed KEY=VALUE pair: %s" str) in
+  let parse str =
+    match String.index_opt str '=' with
+    | None -> error str
+    | Some pos ->
+        let key = String.sub str 0 pos in
+        if not (Re.Pcre.pmatch ~rex:key_re key) then error str
+        else
+          let value = String.sub str (pos + 1) (String.length str - pos - 1) in
+          Ok (key, value)
+  in
+  let print fmt (key, value) = Format.fprintf fmt "%s=%s" key value in
+  Arg.conv' ~docv:"KEY=VALUE" (parse, print)
+
+let env_term : (string * string) list Term.t =
+  let info =
+    Arg.info [ "e"; "env" ] ~docv:"KEY=VALUE"
+      ~doc:
+        "Pass a key/value pair to the function that creates the test suite. \
```

Comment:
how do you pass multiple key value? I see (string * string) list. You use comma to separate them?

Reply:
Oh, I didn't realize the `--help` message didn't make it clear. You'd pass multiple `-e` options, one per key/value pair. Updating the docs.